### PR TITLE
Propagate assistant_id to all events' metadata

### PR DESCRIPTION
**Pull request by ancalita**
**Proposed changes**:
- Implements [ATO-682 subtask](https://rasahq.atlassian.net/browse/ATO-682)

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [X] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/engine/recipes/default_recipe.py [hunk 1]
```diff
@@ -220,6 +220,7 @@ def graph_config_for_recipe(
             train_schema=GraphSchema(train_nodes),
             predict_schema=GraphSchema(predict_nodes),
             training_type=training_type,
+            assistant_id=config.get("assistant_id"),
```
Comment (radovanZRasa) [inline rasa/engine/recipes/default_recipe.py:223]: This is a small thing, but we should start to gradually add small unit tests which would protect us from accidentally changing something like this.

File: rasa/engine/recipes/graph_recipe.py [hunk 2]
```diff
@@ -71,6 +71,7 @@ def graph_config_for_recipe(
             train_schema=GraphSchema.from_dict(config.get("train_schema")),
             predict_schema=GraphSchema.from_dict(config.get("predict_schema")),
             training_type=training_type,
+            assistant_id=config.get("assistant_id"),
```
Comment (radovanZRasa) [inline rasa/engine/recipes/graph_recipe.py:74]: Same here

File: tests/engine/storage/test_storage.py [hunk 3]
```diff
@@ -47,10 +47,12 @@ def test_metadata_serialization(domain: Domain, tmp_path: Path):
     trained_at = datetime.utcnow()
     rasa_version = rasa.__version__
     model_id = "some unique model id"
+    assistant_id = "test_assistant"
     metadata = ModelMetadata(
         trained_at,
         rasa_version,
         model_id,
+        assistant_id,
```
Comment (radovanZRasa) [inline tests/engine/storage/test_storage.py:55]: We need to test if this is set as we are currently doing for rest of parameters.

File: rasa/engine/recipes/default_recipe.py (hunk 1 above)
Comment (ancalita) [reply rasa/engine/recipes/default_recipe.py:223]: Could you please clarify what needs to be protected from accidental changes? The constant name?

File: tests/engine/storage/test_storage.py (hunk 3 above)
Comment (ancalita) [reply tests/engine/storage/test_storage.py:55]: Addressed: https://github.com/RasaHQ/rasa/pull/12019/commits/6ece56025282dbcb0c62b3136be33cef96566f16

File: rasa/engine/recipes/default_recipe.py (hunk 1 above)
Comment (ancalita) [reply rasa/engine/recipes/default_recipe.py]: Added a unit test for this method, let me know if this is what you had in mind: https://github.com/RasaHQ/rasa/pull/12019/commits/6ece56025282dbcb0c62b3136be33cef96566f16

Comment (radovanZRasa) [reply rasa/engine/recipes/default_recipe.py]: We should test if the assignment was done properly. We can add tests for this gradually.

**Review by radovanZRasa — APPROVED** [review]

**Pull request merged**